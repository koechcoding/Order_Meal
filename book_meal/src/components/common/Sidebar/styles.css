.controls {
    height: 100vh;
    min-height: 790px;
    background-color: #eee; }
    .controls .btn {
      display: block;
      width: 80%;
      margin-top: 4.5vh;
      margin-left: 10%;
      transition: 0.1s all; }
    .controls .btn:hover {
      color: black;
      background-color: orange; }
    .controls .btn-logout,
    .controls .btn-logout:hover {
      font-size: 18px;
      color: white;
      padding: 10px;
      border: none;
      color: white;
      background-color: rgba(0, 0, 0, 0.5); }
    .controls .btn.active {
      background-color: orange; }
    .controls hr {
      width: 60%;
      display: block;
      border-top: 2px solid #ddd; }
    .controls .logo {
      text-align: center;
      display: inline-block; }
  
  .controls-icon {
    width: 40px;
    margin-left: calc(50% - 20px); }
  
  .controls.sidebar {
    position: fixed; }