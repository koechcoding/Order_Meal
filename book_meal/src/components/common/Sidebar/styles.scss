.controls {
    height: 100vh;
    min-height: 790px;
    background-color: #eee;

    .btn {
        display: block;
        width: 80%;
        margin-top: 4.5vh;
        margin-left: 10%;
        transition: 0.1s all;
    }

    .btn:hover {
        color: black;
        background-color: orange;
    }

    .btn-logout,
    .btn-logout:hover {
        font-size: 18px;
        color: white;
        padding: 10px;
        border: none;
        color: white;
        background-color: rgba(0, 0, 0, 0.5);
    }

    .btn.active {
        background-color: orange;
    }

    hr {
        width: 60%;
        display: block;
        border-top: 2px solid #ddd;
    }

    .logo {
        text-align: center;
        display: inline-block;
    }
}

.controls-icon {
    width: 40px;
    margin-left: calc(50% - 20px);
}

.controls.sidebar {
    position: fixed;
}