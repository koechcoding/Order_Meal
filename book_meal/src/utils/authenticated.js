export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}