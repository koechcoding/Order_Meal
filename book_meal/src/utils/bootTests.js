import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
