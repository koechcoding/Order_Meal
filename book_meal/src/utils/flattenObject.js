export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}