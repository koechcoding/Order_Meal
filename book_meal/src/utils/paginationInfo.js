export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}