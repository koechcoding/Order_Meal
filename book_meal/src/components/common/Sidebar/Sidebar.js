import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Sidebar = ({children, history}) => (
    <nav className="d-none d-md-block col-md-4 col-lg-3 controls sidebar patterns bg-teal text-center"> 
        <h1 className="logo-medium text-flav pt-3">BAM!</h1> 
        <hr/>
        {children}
        <hr className="last"/>
        <button onClick={() => {localStorage.clear(); history.push('/'); }} className="btn btn-logout">Logout</button>
    </nav>
);

Sidebar.propTypes = {
    children: PropTypes.array,
    history: PropTypes.object
};

export default Sidebar;