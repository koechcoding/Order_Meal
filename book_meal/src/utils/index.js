import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}