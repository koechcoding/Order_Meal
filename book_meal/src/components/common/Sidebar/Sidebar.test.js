
import React from 'react';
import sinon from 'sinon';
import 'src/utils/bootTests'; 
import Sidebar from './Sidebar';
import { shallow }  from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('<Sidebar />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Sidebar history={{push: () => {}}} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('logs out a user', () => {
        const spy = sinon.fake();
        const wrapper = shallow(<Sidebar history={{push: spy}} />)
        wrapper.find('.btn-logout').simulate('click');
        expect(spy.called).toBe(true);
    })
});
