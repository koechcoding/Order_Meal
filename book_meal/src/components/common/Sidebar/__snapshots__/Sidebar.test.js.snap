// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="logo-medium text-flav pt-3"
>
          BAM!
</h1>,
        <hr />,
        undefined,
        <hr
          className="last"
/>,
        <button
          className="btn btn-logout"
          onClick={[Function]}
>
          Logout
</button>,
      ],
      "className": "d-none d-md-block col-md-4 col-lg-3 controls sidebar patterns bg-teal text-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BAM!",
          "className": "logo-medium text-flav pt-3",
        },
        "ref": null,
        "rendered": "BAM!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "last",
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "className": "btn btn-logout",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Logout",
        "type": "button",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="logo-medium text-flav pt-3"
>
            BAM!
</h1>,
          <hr />,
          undefined,
          <hr
            className="last"
/>,
          <button
            className="btn btn-logout"
            onClick={[Function]}
>
            Logout
</button>,
        ],
        "className": "d-none d-md-block col-md-4 col-lg-3 controls sidebar patterns bg-teal text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BAM!",
            "className": "logo-medium text-flav pt-3",
          },
          "ref": null,
          "rendered": "BAM!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "last",
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "className": "btn btn-logout",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logout",
          "type": "button",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;